gluten-core/src/main/scala/io/glutenproject/expression/InSetOperatorTransformer.scala

```diff
@@ -58,57 +59,104 @@ object InSetOperatorTransformer {
       case _: ByteType =>
         new util.ArrayList[ExpressionNode](values.map({
           value =>
-            ExpressionBuilder.makeByteLiteral(value.asInstanceOf[java.lang.Byte])
+            if (value == null) {
```

Comment:
Looks like all this part of code can be simplified by reusing below function.
https://github.com/oap-project/gluten/blob/main/gluten-core/src/main/java/io/glutenproject/substrait/expression/ExpressionBuilder.java#L125
Then it can become sth. like below.
```
 new util.ArrayList[ExpressionNode](values.map({
    value =>
      ExpressionBuilder.makeLiteral(value, type, nullable)
  }).asJava)
```

Reply:
You are right definitely.  I fixed it.

Reply:
Thanks. Please also compare the two parts and add the missing type into `ExpressionBuilder.makeLiteral`. For example, timestamp is here but not in `ExpressionBuilder.makeLiteral`.

Reply:
1. Add float and timestamp type.
2. reorder the type sequence.(start with bool).